### DISCUSSION TITLE
ALM terminated unexpectedly

### DISCUSSION THREAD

**Discussion by Lamed-git** [ttadano/alamode#220] [Q&A] [answered]
Dear Dr. Tadano,
I use the latest version of ALAMODE (1.5.0), but I still encounter the ALM unexpected termination situation, I try to change FCSYM_BASIS to C, but the error still exists, the same supercell training the second order force constant does not report an error, but when I get the anharmonic interatomic force constants something goes wrong:
![fff31c0ebe2d5532063734ebaefb3a9](https://github.com/user-attachments/assets/941a298a-147f-41ed-8545-15aac15dc700)
I checked the input file carefully to make sure that the content you mentioned earlier is OK：
1. ALM is able to correctly identify symmetries (No.164 space group), and 16 decimal places are used for both cell parameters and atomic sites.
2. The system itself has 108 atoms, 199 input files are used for training, the total free parameter is 29076, the memory used is 256 GB, there should be no memory overflow problem.
3. I adjusted the FCSYM_BASIS (L or C) and LMODEL (enet or adaptive-lasso), but it didn't help, the output still terminated at:
![image](https://github.com/user-attachments/assets/9951de41-1ba7-4017-ba9b-0fddc48c5796)

**Comment by ttadano** [comment]
Hi,

This issue is probably related to the numerical instability of the matrix reduction algorithm inside ALM, but please let me confirm this. 

For the moment, please try setting `TOL_CONST` to larger or smaller than 1.0e-6 (default), which may change the behavior of the code.

**Reply by Lamed-git** [reply]
Hi,
I can't find **TOL_CONST** in the List of supported input variables, only **TOLERANCE** (default value: 1.0e-3) and **CONV_TOL** (default value: 1.0e-8) seem to be available at the moment. I tried to increase **CONV_TOL** to 1.0e-3 and ALM worked well. I'm not sure that's a reasonable adjustment.

**Reply by ttadano** [reply, marked as answer]
`TOL_CONST` is not included on the webpage, but it is available. 

You can change it by writing, for example, `TOL_CONST = 1.0e-8` in the `&general` field.

**Reply by Lamed-git** [reply]
I've solved this problem by change TOL_CONST = 1.0e-8.

**Reply by ttadano** [reply]
I'm delighted to hear that. Still, I will work on improving the stability of the algorithm.
